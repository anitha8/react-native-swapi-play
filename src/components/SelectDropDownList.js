/**
 * @flow
 */
import React, { Component } from "react";
import {
    Text,
    View,
    TextInput,
    TouchableWithoutFeedback,
    FlatList,
    Image, StyleSheet,
} from "react-native";
import PropTypes from "prop-types";
import find from "lodash/find";
import get from "lodash/get";
import ListItem from "./ListItem";
import usernameImg from '../images/username.png';
import passwordImg from '../images/password.png';

class SelectDropDownList extends Component {
    static propTypes = {
        selectedItems: PropTypes.array,
        items: PropTypes.array.isRequired,
        uniqueKey: PropTypes.string,
        onSelectedItemsChange: PropTypes.func.isRequired,
        placeHolderText: PropTypes.string,
        displayKey: PropTypes.string,
        getSelectedItems: PropTypes.func,
    };

    static defaultProps = {
        selectedItems: [],
        items: [],
        uniqueKey: "id",
        onSelectedItemsChange: () => {
        },
        placeHolderText: "Select",
        displayKey: "name",
    };

    constructor(props) {
        super(props);
        this.state = {
            selector: false,
            searchTerm: "",
            textColor: '#f0f1f1'
        };
    }

    findItem = itemKey => {
        const { items, uniqueKey } = this.props;
        return find(items, singleItem => singleItem[uniqueKey] === itemKey) || {};
    };

    onChangeInput = value => {
        this.setState({ searchTerm: value });
    };

    getSelectLabel = () => {
        const {
            placeHolderText,
            selectedItems,
            displayKey,
        } = this.props;

        const item = selectedItems[0];
        const foundItem = this.findItem(item);
        const getItemname = get(foundItem, displayKey);
        return getItemname || placeHolderText;
    };

    toggleSelector = () => {
        this.setState({
            selector: !this.state.selector,
        });
        const {
            selectedItems,
            displayKey,
            placeHolderText,
        } = this.props;
        const item = selectedItems[0];
        const foundItem = this.findItem(item);
        const getItemname = get(foundItem, displayKey);
        this.setState({ textColor: getItemname != placeHolderText && 'black' || '#f0f1f1' });
    };

    submitSelection = () => {
        this.setState({ searchTerm: "" });
        this.toggleSelector();
    };

    itemSelected = item => {
        const { uniqueKey, selectedItems } = this.props;
        return !!find(selectedItems, singleItem => item[uniqueKey] === singleItem);
    };

    toggleItem = item => {
        const {
            uniqueKey,
            onSelectedItemsChange,
        } = this.props;

        this.submitSelection();
        onSelectedItemsChange([item[uniqueKey]]);
    };

    getRow = item => {
        const { displayKey } = this.props;
        const selectedItemFlag = this.itemSelected(item);

        return (
            <ListItem
                selectedItemFlag={selectedItemFlag}
                onPress={() => this.toggleItem(item)}
                itemText={item[displayKey]}
                gradientColor={item.colorVal}
            />
        );
    };

    filterItems = searchTerm => {
        const { items, displayKey } = this.props;
        const filteredItems = [];
        items.forEach(item => {
            const parts = searchTerm.trim().split(/[ \-:]+/);
            const regex = new RegExp(`(${parts.join("|")})`, "ig");
            if (regex.test(get(item, displayKey))) {
                filteredItems.push(item);
            }
        });
        return filteredItems;
    };

    renderItems = () => {
        const { items, uniqueKey, selectedItems } = this.props;
        const { searchTerm } = this.state;
        let component = null;
        const renderItems = searchTerm
            ? this.filterItems(searchTerm.trim())
            : items;
        if (renderItems.length) {
            component = (
                <FlatList
                    data={renderItems}
                    keyExtractor={item => item[uniqueKey]}
                    renderItem={rowData => this.getRow(rowData.item)}
                />
            );
        }
        return component;
    };

    render() {
        const { selector, textColor } = this.state;
        const { placeHolderText } = this.props;
        let listStyle = selector ? styles.selectorView : styles.dropdownView;
        let text;
        let items;

        if (selector) {
            text = <TextInput
                onChangeText={this.onChangeInput}
                placeholder={placeHolderText}
                placeholderTextColor={'#f0f1f1'}
                underlineColorAndroid="transparent"
                style={styles.container}
            />;
        } else {
            text = <Text
                style={[styles.container, {color: textColor}]}
                >{this.getSelectLabel()}
                </Text>;
        }

        if (selector) {
            items = <View>{this.renderItems()}</View>;
        }

        return (
            <View style={listStyle}>
                {<TouchableWithoutFeedback onPress={this.toggleSelector}>
                    <View style={styles.inputGroup}>
                        {text}
                        <View style={styles.dropdownIcon}>
                            <Image
                                source={selector ? usernameImg : passwordImg}
                                style={styles.indicator}
                            />
                        </View>
                    </View>
                </TouchableWithoutFeedback>}
                {items}
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    indicator: {
        height: 10,
        width: 20,
    },
    dropdownIcon: {
        paddingTop: 10,
        paddingLeft: 10,
        paddingRight: 10,
        paddingBottom: 10,
    },
    selectedItem: {
        flexDirection: "row",
        alignItems: "center",
        paddingLeft: 15,
        margin: 3,
        borderRadius: 20,
        borderWidth: 2,
        justifyContent: "center",
        height: 40,
    },
    selectedItemText: {
        flex: 1,
        fontSize: 15,
    },
    selectorView: {
        flexDirection: "column",
        marginBottom: 10,
        elevation: 2,
    },
    inputGroup: {
        flexDirection: "row",
        alignItems: "center",
        paddingLeft: 16,
        backgroundColor: 'white',
        height: 40,
        borderBottomWidth: 1,
        borderBottomColor: 'black',
    },
    dropdownView: {
        flexDirection: "column",
        alignItems: "center",
        height: 40,
        marginBottom: 10,
    },
    menuListBox: {
        flexDirection: "row",
        flexWrap: "wrap",
    },
});

export default SelectDropDownList;

