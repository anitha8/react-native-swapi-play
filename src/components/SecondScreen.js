import React, { Component, PropTypes } from 'react';
import {
	StyleSheet,
	View,
	Image,
	TouchableOpacity,
	Animated,
	Easing,
	Text,
} from 'react-native';
import Dimensions from 'Dimensions';

import { Actions } from 'react-native-router-flux';
import SelectDropDownList from "./SelectDropDownList";
import arrowImg from '../images/left-arrow.png';

const SIZE = 40;
var count = 0;
const SWAPIConfig = {
    baseUrl: 'http://swapi.co/api/',
    planetsEndpoint: 'planets/',
};
var REQUEST_URL = SWAPIConfig.baseUrl + SWAPIConfig.planetsEndpoint;

export default class SecondScreen extends Component {
	constructor() {
		super();

		this.state = {
            isLoading: false,
            selectedItems: [],
            items: [],
            completeData: [],
            completeSelectedData: [],
        };


		this._onPress = this._onPress.bind(this);
		this.growAnimated = new Animated.Value(0);
	}

	componentDidMount(){
        count = 0;
        this.fetchPlanetData();
    }
	_onPress() {
		if (this.state.isLoading) return;

		this.setState({ isLoading: true });

		Animated.timing(
			this.growAnimated,
			{
				toValue: 1,
				duration: 300,
				easing: Easing.linear,
			}
		).start();

		setTimeout(() => {
			Actions.pop();
		}, 500);
	}

    onSelectedItemsChange = selectedItems => {
		const { completeData, completeSelectedData } = this.state;
		this.setState({completeSelectedData: completeData[selectedItems]});
        this.setState({ selectedItems });
    };

    fetchPlanetData() {
        let planetData = [];
        fetch(REQUEST_URL)
            .then((response) => response.json())
            .then((responseData) => {
            	this.setState({completeData: responseData.results});
            	for(i=0;i<responseData.results.length;i++){
					let population = parseInt(responseData.results[i].population, 10) || 0;

                    planetData.push({
                        id: i,
                        name: responseData.results[i].name,
                        population: population,
                    })
				}

				planetData.sort((a, b) => parseInt(a.population, 10) < parseInt(b.population, 10));

                for(i=0;i<planetData.length;i++) {
                    let colorVal;

                    if (planetData[i].population > 0) {
                        count = count + 10;
                        planetData[i].colorVal = this.colorLuminance("FF0000", count);
                    } else {
                        planetData[i].colorVal = '#ffffff';
                    }
                }
                    this.setState({
                    items: planetData,
                    isLoading: true
                });
            })
            .done();
    }


    rgbToHsl(r, g, b){
        r /= 255, g /= 255, b /= 255;
        var max = Math.max(r, g, b), min = Math.min(r, g, b);
        var h, s, l = (max + min) / 2;

        if(max == min){
            h = s = 0; // achromatic
        }else{
            var d = max - min;
            s = l > 0.5 ? d / (2 - max - min) : d / (max + min);
            switch(max){
                case r: h = (g - b) / d + (g < b ? 6 : 0); break;
                case g: h = (b - r) / d + 2; break;
                case b: h = (r - g) / d + 4; break;
            }
            h /= 6;
        }

        return [h, s, l];
    }

    hslToRgb(h, s, l){
        var r, g, b;

        if(s == 0){
            r = g = b = l; // achromatic
        }else{
            function hue2rgb(p, q, t){
                if(t < 0) t += 1;
                if(t > 1) t -= 1;
                if(t < 1/6) return p + (q - p) * 6 * t;
                if(t < 1/2) return q;
                if(t < 2/3) return p + (q - p) * (2/3 - t) * 6;
                return p;
            }

            var q = l < 0.5 ? l * (1 + s) : l + s - l * s;
            var p = 2 * l - q;
            r = hue2rgb(p, q, h + 1/3);
            g = hue2rgb(p, q, h);
            b = hue2rgb(p, q, h - 1/3);
        }

        return [r * 255, g * 255, b * 255];
    }

    colorLuminance(rgbcode, percent) {
        var r = parseInt(rgbcode.slice(1, 3), 16),
            g = parseInt(rgbcode.slice(3, 5), 16),
            b = parseInt(rgbcode.slice(5, 7), 16),
            HSL = this.rgbToHsl(r, g, b),
            newBrightness = HSL[2] + HSL[2] * (percent / 100),
            RGB;

        RGB = this.hslToRgb(HSL[0], HSL[1], newBrightness);
        rgbcode = '#'
            + this.convertToTwoDigitHexCodeFromDecimal(RGB[0])
            + this.convertToTwoDigitHexCodeFromDecimal(RGB[1])
            + this.convertToTwoDigitHexCodeFromDecimal(RGB[2]);

        return rgbcode;
    }

    convertToTwoDigitHexCodeFromDecimal(decimal){
        var code = Math.round(decimal).toString(16);

        (code.length > 1) || (code = '0' + code);
        return code;
    }

    _onPress() {
		Actions.pop();
    }
    render() {
        const { items, selectedItems, completeSelectedData } = this.state;

		const changeScale = this.growAnimated.interpolate({
			inputRange: [0, 1],
			outputRange: [1, SIZE],
		});
		return (
			<View style={styles.container}>
                <SelectDropDownList
                    items={items}
                    onSelectedItemsChange={this.onSelectedItemsChange}
                    selectedItems={selectedItems}
                    placeHolderText="Type planet name"
                />
				<Text style={{fontSize: 16, color: '#111',paddingLeft:10, borderWidth:1, borderColor:"#111" }}>
					{completeSelectedData.population + '\n' + completeSelectedData.climate }
				</Text>
                <Animated.View style={{width: DEVICE_WIDTH - MARGIN}}>
					<TouchableOpacity style={styles.button}
									   onPress={this._onPress}
									   activeOpacity={1} >
						<Text style={styles.text}>LOG OUT</Text>
					</TouchableOpacity>
                </Animated.View>
			</View>
		);
	}
}

const DEVICE_WIDTH = Dimensions.get('window').width;
const DEVICE_HEIGHT = Dimensions.get('window').height;
const MARGIN = 40;
const styles = StyleSheet.create({
	container: {
		flex: 1,
	},
	circle: {
		height: SIZE,
		width: SIZE,
		marginTop: -SIZE,
		borderRadius: 100,
		backgroundColor: '#ADD8E6',
	},
	image: {
		width: 24,
		height: 24,
	},
    button: {
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#ADD8E6',
        height: MARGIN,
        borderRadius: 20,
        zIndex: 100,
		marginLeft: 25,
        marginTop: 100,
    },
});