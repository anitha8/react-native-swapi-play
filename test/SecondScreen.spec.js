import 'react-native';
import React from 'react';
import SecondScreen from '../src/components/SecondScreen';
import SelectDropDownList from "../src/components/SelectDropDownList";

import {configure, shallow} from "enzyme/build/index";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("Testing SelectDropDownList Component", () => {

    beforeEach(() => {
        wrapper = shallow(<SecondScreen/>);
    });

    it("Check SelectDropDownList Snapshot", () => {
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it("check planet search visiable", () => {
        wrapper = shallow(<SecondScreen/>);
        const render = wrapper.dive();
        expect(render.find(SelectDropDownList).length).toBe(1);
    });
});