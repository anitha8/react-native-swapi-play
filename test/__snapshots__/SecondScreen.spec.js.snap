// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SelectDropDownList Component Check SelectDropDownList Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageBackground
    source={1}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "marginTop": 20,
          },
        ]
      }
    >
      <SelectDropDownList
        displayKey="name"
        items={Array []}
        onSelectedItemsChange={[Function]}
        placeHolderText="Type planet name"
        selectedItems={Array []}
        uniqueKey="id"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "borderColor": "#111",
            "borderWidth": 1,
            "color": "#111",
            "fontSize": 16,
            "paddingLeft": 10,
          }
        }
      >
        Population: undefined

Climate: undefined

Terrain: undefined
      </Text>
      <AnimatedComponent
        style={
          Object {
            "width": 710,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={1}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ADD8E6",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 25,
              "marginTop": 100,
              "zIndex": 100,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={undefined}
          >
            LOG OUT
          </Text>
        </TouchableOpacity>
      </AnimatedComponent>
    </View>
  </ImageBackground>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />,
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "marginTop": 20,
              },
            ]
          }
        >
          <SelectDropDownList
            displayKey="name"
            items={Array []}
            onSelectedItemsChange={[Function]}
            placeHolderText="Type planet name"
            selectedItems={Array []}
            uniqueKey="id"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "borderColor": "#111",
                "borderWidth": 1,
                "color": "#111",
                "fontSize": 16,
                "paddingLeft": 10,
              }
            }
          >
            Population: undefined

Climate: undefined

Terrain: undefined
          </Text>
          <AnimatedComponent
            style={
              Object {
                "width": 710,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={1}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ADD8E6",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 25,
                  "marginTop": 100,
                  "zIndex": 100,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={undefined}
              >
                LOG OUT
              </Text>
            </TouchableOpacity>
          </AnimatedComponent>
        </View>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": 1,
          "style": Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <SelectDropDownList
              displayKey="name"
              items={Array []}
              onSelectedItemsChange={[Function]}
              placeHolderText="Type planet name"
              selectedItems={Array []}
              uniqueKey="id"
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderColor": "#111",
                  "borderWidth": 1,
                  "color": "#111",
                  "fontSize": 16,
                  "paddingLeft": 10,
                }
              }
            >
              Population: undefined

Climate: undefined

Terrain: undefined
            </Text>,
            <AnimatedComponent
              style={
                Object {
                  "width": 710,
                }
              }
            >
              <TouchableOpacity
                activeOpacity={1}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ADD8E6",
                    "borderRadius": 20,
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 25,
                    "marginTop": 100,
                    "zIndex": 100,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={undefined}
                >
                  LOG OUT
                </Text>
              </TouchableOpacity>
            </AnimatedComponent>,
          ],
          "style": Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginTop": 20,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "displayKey": "name",
              "items": Array [],
              "onSelectedItemsChange": [Function],
              "placeHolderText": "Type planet name",
              "selectedItems": Array [],
              "uniqueKey": "id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Population: undefined

Climate: undefined

Terrain: undefined",
              "disabled": false,
              "ellipsizeMode": "tail",
              "style": Object {
                "borderColor": "#111",
                "borderWidth": 1,
                "color": "#111",
                "fontSize": 16,
                "paddingLeft": 10,
              },
            },
            "ref": null,
            "rendered": "Population: undefined

Climate: undefined

Terrain: undefined",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <TouchableOpacity
                activeOpacity={1}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ADD8E6",
                    "borderRadius": 20,
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 25,
                    "marginTop": 100,
                    "zIndex": 100,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={undefined}
                >
                  LOG OUT
                </Text>
              </TouchableOpacity>,
              "style": Object {
                "width": 710,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 1,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={undefined}
                >
                  LOG OUT
                </Text>,
                "onPress": [Function],
                "style": Object {
                  "alignItems": "center",
                  "backgroundColor": "#ADD8E6",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 25,
                  "marginTop": 100,
                  "zIndex": 100,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "LOG OUT",
                  "disabled": false,
                  "ellipsizeMode": "tail",
                  "style": undefined,
                },
                "ref": null,
                "rendered": "LOG OUT",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={1}
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                undefined,
              ]
            }
          />,
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "marginTop": 20,
                },
              ]
            }
          >
            <SelectDropDownList
              displayKey="name"
              items={Array []}
              onSelectedItemsChange={[Function]}
              placeHolderText="Type planet name"
              selectedItems={Array []}
              uniqueKey="id"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderColor": "#111",
                  "borderWidth": 1,
                  "color": "#111",
                  "fontSize": 16,
                  "paddingLeft": 10,
                }
              }
            >
              Population: undefined

Climate: undefined

Terrain: undefined
            </Text>
            <AnimatedComponent
              style={
                Object {
                  "width": 710,
                }
              }
            >
              <TouchableOpacity
                activeOpacity={1}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ADD8E6",
                    "borderRadius": 20,
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 25,
                    "marginTop": 100,
                    "zIndex": 100,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={undefined}
                >
                  LOG OUT
                </Text>
              </TouchableOpacity>
            </AnimatedComponent>
          </View>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <SelectDropDownList
                displayKey="name"
                items={Array []}
                onSelectedItemsChange={[Function]}
                placeHolderText="Type planet name"
                selectedItems={Array []}
                uniqueKey="id"
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "borderColor": "#111",
                    "borderWidth": 1,
                    "color": "#111",
                    "fontSize": 16,
                    "paddingLeft": 10,
                  }
                }
              >
                Population: undefined

Climate: undefined

Terrain: undefined
              </Text>,
              <AnimatedComponent
                style={
                  Object {
                    "width": 710,
                  }
                }
              >
                <TouchableOpacity
                  activeOpacity={1}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ADD8E6",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 25,
                      "marginTop": 100,
                      "zIndex": 100,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={undefined}
                  >
                    LOG OUT
                  </Text>
                </TouchableOpacity>
              </AnimatedComponent>,
            ],
            "style": Array [
              Object {
                "flex": 1,
              },
              Object {
                "marginTop": 20,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "displayKey": "name",
                "items": Array [],
                "onSelectedItemsChange": [Function],
                "placeHolderText": "Type planet name",
                "selectedItems": Array [],
                "uniqueKey": "id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Population: undefined

Climate: undefined

Terrain: undefined",
                "disabled": false,
                "ellipsizeMode": "tail",
                "style": Object {
                  "borderColor": "#111",
                  "borderWidth": 1,
                  "color": "#111",
                  "fontSize": 16,
                  "paddingLeft": 10,
                },
              },
              "ref": null,
              "rendered": "Population: undefined

Climate: undefined

Terrain: undefined",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TouchableOpacity
                  activeOpacity={1}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ADD8E6",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 25,
                      "marginTop": 100,
                      "zIndex": 100,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={undefined}
                  >
                    LOG OUT
                  </Text>
                </TouchableOpacity>,
                "style": Object {
                  "width": 710,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeOpacity": 1,
                  "children": <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={undefined}
                  >
                    LOG OUT
                  </Text>,
                  "onPress": [Function],
                  "style": Object {
                    "alignItems": "center",
                    "backgroundColor": "#ADD8E6",
                    "borderRadius": 20,
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 25,
                    "marginTop": 100,
                    "zIndex": 100,
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "LOG OUT",
                    "disabled": false,
                    "ellipsizeMode": "tail",
                    "style": undefined,
                  },
                  "ref": null,
                  "rendered": "LOG OUT",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
